testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/account/AccountRestServiceTest.java

```diff
@@ -304,6 +306,35 @@ public void listApplications() throws IOException {
                 .asJson(new TypeReference<List<ClientRepresentation>>() {
                 });
         assertFalse(applications.isEmpty());
+
+        Map<String, ClientRepresentation> apps = applications.stream().collect(Collectors.toMap(x -> x.getClientId(), x -> x));
+        Assert.assertThat(apps.keySet(), containsInAnyOrder("account", "admin-cli", "broker", "security-admin-console", "test-app", "root-url-client", "test-app-scope", "third-party", "test-app-authz", "named-test-app", "var-named-test-app", "direct-grant"));
+
+        ClientRepresentation accountApp = apps.get("account");
+        assertEquals(  "/auth/realms/test/account", accountApp.getUrl());
+
+        ClientRepresentation testApp = apps.get("test-app");
+        assertEquals("http://localhost:8180/auth/realms/master/app/auth", testApp.getUrl());
+
+        ClientRepresentation thirdPartyApp = apps.get("third-party");
+        assertEquals("A third party application", thirdPartyApp.getDescription());
+        assertEquals("http://localhost:8180/auth/realms/master/app/auth", thirdPartyApp.getUrl());
+
+        ClientRepresentation testAppNamedApp = apps.get("var-named-test-app");
+        assertEquals("http://localhost:8180/varnamedapp/base", testAppNamedApp.getUrl());
+
+        ClientRepresentation rootUrlClientApp = apps.get("root-url-client");
+        assertEquals("http://localhost:8180/foo/bar/baz", rootUrlClientApp.getUrl());
+
+        ClientRepresentation authzApp = apps.get("test-app-authz");
+        assertEquals("/test-app-authz", authzApp.getUrl());
+
+
+        ClientRepresentation namedApp = apps.get("named-test-app");
+        assertEquals("http://localhost:8180/namedapp/base", namedApp.getUrl());
+
+        ClientRepresentation appScopeApp = apps.get("test-app-scope");
+        assertNull(appScopeApp.getUrl());
```

Comment:
We need to test all the fields, not just `url`. This also includes the "generated" fields like `scopes`, `inUse`, `internal`, etc.